docker-compose.yml

```diff
@@ -0,0 +1,83 @@
+version: '3.9'
+
+services:
+  server:
+    image: grasscutter:1.0
+    build: .
+    volumes:
+      - ./resources:/app/resources
+      - ./data:/app/data
+      - ./config.json:/app/config.json
+      - ./keystore.p12:/app/keystore.p12
+    network_mode: service:mongodb
+    depends_on:
+      - mongodb
+    stdin_open: true
+    tty: true
+  
+  mongodb:
+    image: mongo:5.0.7
+    ports:
+      - 80:80
+      - 443:443
+      - 8888:8888/tcp
+      - 8888:8888/udp
+      - 27017:27017
+      - 22102:22102/tcp
+      - 22102:22102/udp
+    volumes:
+      - ./docker_data/mongo:/data/db
+
+  config-gen:
+    image: grasscutter:1.0
+    build: .
+    volumes:
+      - ./:/app/src
+    entrypoint: ["/bin/sh", "-c", "
+      echo [!!] config.json will be deleted;
+      echo - Any key to continue;
+      echo - Ctrl+C  to abort;
+      read answer;
+
+      java -jar grasscutter.jar;
+      rm -rf ./src/config.json;
+      mv config.json ./src;
+      echo config generated"]
+    stdin_open: true
+    tty: true
+    profiles:
+      - donotstart
+
+  res-gen:
+    image: alpine/git:v2.34.1
+    volumes:
+      - ./:/app/src
+    entrypoint: ["/bin/sh", "-c", "
+      echo [!!] resources folder will be deleted;
+      echo - Any key to continue;
+      echo - Ctrl+C  to abort;
+      read answer;
+
+      echo Deleting old resources...;
+      rm -rf /app/src/resources;
+      
+      echo Downloading resources...;
+      git clone --depth 1 https://github.com/Dimbreath/GenshinData.git /tmp/GenshinData;
+      git clone --depth 1 https://github.com/radioegor146/gi-bin-output.git /tmp/gi-bin-output;
+
+      echo Copying resources...;
+      mkdir -p /app/src/resources/;
+      mv /tmp/GenshinData/TextMap /app/src/resources;
+      mv /tmp/GenshinData/Subtitle /app/src/resources;
+      mv /tmp/GenshinData/Readable /app/src/resources;
+      mv /tmp/GenshinData/ExcelBinOutput /app/src/resources;
+      mv /tmp/gi-bin-output/2.5.52/Data/_BinOutput /app/src/resources/BinOutput;
```

Comment:
Since radioegor146's repo got DMCA'd, you need to use different data from somewhere else.

Reply:
Thanks, fixed.